#143412453 Limit Number Of Players

andela-wmumbi opened the pull request:
#### What does this PR do?
Users should be able to invite people to their game up to a maximum of 11 other people, so no game can exceed 12 people in total and must have at least 3 players.
 #### Description of Task to be completed?
bower install sweetalert
 #### How should this be manually tested?
run gulp
 #### What are the relevant pivotal tracker stories?
#143412453
 #### Screenshots (if appropriate)
<img width="1280" alt="game" src="https://cloud.githubusercontent.com/assets/27006277/25388460/9b0d014a-29d6-11e7-9675-dd05df30d43c.png">
<img width="1280" alt="max" src="https://cloud.githubusercontent.com/assets/27006277/25495829/62a822c8-2b87-11e7-8c1f-e236cfee8891.png">

andela-wmumbi committed:
[Feature #[143412453]]send invites to at most 11 people

andela-wmumbi committed:
[Feature #[143412453]]send invites to at most 11 people

gngchrs commented:
Few Points:

- Resovle merge conflicts
- Check PR naming convention here from the wiki https://github.com/andela/reactors-cards-for-humanity/wiki/PR-naming-and-Description
- You only have the less than three people screenshot, add the more than 11 players one.

gngchrs commented on app/views/includes/foot.jade:
```diff
@@ -13,6 +13,7 @@ script(type='text/javascript', src='/lib/bootstrap/dist/js/bootstrap.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular-resource.js')
 script(type='text/javascript', src='https://code.angularjs.org/1.1.5/angular-cookies.js')
+//script(type='text/javascript', src='/lib/angular-toastr/dist/angular-toastr.min.js')
```
minor: delete this line, since it's may never be used

gngchrs commented on bower.json:
```diff
@@ -6,7 +6,9 @@
-    "underscore": "~1.5.2"
+    "underscore": "~1.5.2",
+    "angular-toastr": "^2.1.1",
+    "angular-animate": "^1.6.4"
```
minor: delete these dependencies. We are not using them

gngchrs commented on package.json:
```diff
@@ -40,6 +40,7 @@
     "gulp-mocha": "^4.3.0",
     "gulp-nodemon": "^2.2.1",
     "gulp-sass": "^3.1.0",
+    "install": "^0.8.8",
```
what does this do?

gngchrs commented on package.json:
```diff
@@ -57,6 +58,7 @@
     "sinon": "~2.1.0",
     "socket.io": "~1.7.3",
     "socket.io-client": "~1.7.3",
+    "sweetalert": "^1.1.3",
```
I'm not sure how your sweetalert worked. And how it got to the `public/lib` folder. This should be in `bower.json` or maybe just get it as a CDN https://cdnjs.com/libraries/sweetalert

gngchrs commented on package.json:
```diff
@@ -65,6 +67,14 @@
     "eslint": "^3.19.0",
     "eslint-config-airbnb-base": "^11.1.3",
     "eslint-plugin-import": "^2.2.0",
+    "gulp": "^3.9.1",
```
You probably need to rebase you branch. Gulp was already installed by @Mercy-Muchai

gngchrs commented on package.json:
```diff
@@ -65,6 +67,14 @@
+    "gulp-mocha": "^4.3.0",
+    "gulp-nodemon": "^2.2.1",
+    "gulp-sass": "^3.1.0",
+    "nodemon": "^1.11.0",
```
ditto ☝️

andela-wmumbi committed:
[Feature #143412453] adding search api
- limits maximum players to twelve per game
- implement api to return all users

LWanjiru commented on config/express.js:
```diff
@@ -46,12 +46,15 @@ module.exports = (app, passport, mongoose) => {
   // bodyParser should be above methodOverride
-  app.use(bodyParser());
+  app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({extended: true}));
```
There should be space between the starting curly brace and the words and also the closing one i.e.  ...({ extended: true }).

LWanjiru commented on config/socket/game.js:
```diff
@@ -185,71 +191,71 @@ Game.prototype.selectFirst = function() {
-  self.state = "waiting for czar to decide";
+Game.prototype.stateJudging = function (self) {
+  self.state = 'waiting for czar to decide';
   // console.log(self.gameID,self.state);
```
This comment doesn't seem to be relevant to the code. It should be removed

LWanjiru commented on config/socket/game.js:
```diff
@@ -132,27 +138,27 @@ Game.prototype.prepareGame = function() {
-  self.state = "waiting for players to pick";
+Game.prototype.stateChoosing = function (self) {
+  self.state = 'waiting for players to pick';
   // console.log(self.gameID,self.state);
```
This comment doesn't seem to be relevant to the code. It should be removed

LWanjiru commented on config/socket/game.js:
```diff
@@ -185,71 +191,71 @@ Game.prototype.selectFirst = function() {
+Game.prototype.stateResults = function (self) {
+  self.state = 'winner has been chosen';
   console.log(self.state);
   // TODO: do stuff
```
What does this comment mean?

LWanjiru commented on config/socket/game.js:
```diff
@@ -382,43 +387,43 @@ Game.prototype.removePlayer = function(thisPlayer) {
+      console.log('WARNING: czar', thisPlayer, 'picked a card that was not on the table.');
     }
   } else {
     // TODO: Do something?
```
What does this comment mean?

LWanjiru commented on config/socket/socket.js:
```diff
@@ -65,71 +65,75 @@ module.exports = function(io) {
         if (err) {
-          console.log('err',err);
+          console.log('err', err);
           return err; // Hopefully this never happens.
```
What does this comment mean?

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-    $scope.showFirst = function(card) {
-      return game.curQuestion.numAnswers > 1 && $scope.pickedCards[0] === card.id;
-    };
+  $scope.showFirst = function (card) {
```
Use arrow function here.

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
+    return false;
+  };
+
+  $scope.secondAnswer = function ($index) {
```
Use arrow function here.

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-        return false;
-      }
-    };
+  $scope.firstAnswer = function ($index) {
```
Use arrow function here.

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-        return false;
-      }
-    };
+  $scope.cardIsSecondSelected = function (card) {
```
use arrow function

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-        return false;
-      }
-    };
+  $scope.cardIsFirstSelected = function (card) {
```
Use arrow function here.

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
+  let makeAWishFacts = MakeAWishFactsService.getMakeAWishFacts();
+  $scope.makeAWishFact = makeAWishFacts.pop();
+
+  $scope.pickCard = function (card) {
```
Use arrow function here.

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-            $scope.sendPickedCards();
-            $scope.hasPickedCards = true;
-          } else if (game.curQuestion.numAnswers === 2 &&
+.controller('GameController', ['$scope', 'game', '$timeout', '$location', 'MakeAWishFactsService', '$dialog', function ($scope, game, $timeout, $location, MakeAWishFactsService, $dialog) {
```
function... ES6

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-        return false;
-      }
-    };
+  $scope.sendPickedCards = function () {
```
This will raise an error when listing

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-    $scope.showSecond = function(card) {
-      return game.curQuestion.numAnswers > 1 && $scope.pickedCards[1] === card.id;
-    };
+  $scope.showSecond = function (card) {
```
consider using arrow function.

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-    $scope.isCzar = function() {
-      return game.czar === game.playerIndex;
-    };
+  $scope.isCzar = function () {
```
unnamed function

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-    $scope.isPlayer = function($index) {
-      return $index === game.playerIndex;
-    };
+  $scope.isPlayer = function ($index) {
```
use arrow function

LWanjiru commented on public/js/controllers/game.js:
```diff
@@ -1,184 +1,181 @@
-        $scope.winningCardPicked = true;
-      }
-    };
+  $scope.pickWinning = function (winningSet) {
```
Use arrow function

LWanjiru commented on testing.js:
```diff
@@ -0,0 +1,31 @@
+/*
+ * A simple React component
```
This comment is not relevant to the code

LWanjiru commented on testing.js:
```diff
@@ -0,0 +1,31 @@
+/*
+ * A simple React component
+ */
+// What is happening here?
```
This comment is not relevant to the code

LWanjiru commented on testing.js:
```diff
@@ -0,0 +1,31 @@
+  constructor(props){
+     super(props);
+
+    // What does the state object do?
```
This comment is not relevant to the code

LWanjiru commented on testing.js:
```diff
@@ -0,0 +1,31 @@
+ * A simple React component
+ */
+// What is happening here?
+class Why extends React.Component {
```
Is this file part of the CFH codebase? It is on react and the codebase uses angular...

LWanjiru reviewed (changes requested):
Needs linting, relevant comments and changes on a file that seems to not belong in the codebase

andela-wmumbi committed:
[feature #143412453] limit max players
-implement api for searching users
-limit maximum number of players

gngchrs commented:
fix linting and merge conflicts then reopen
